### PR TITLE
feat: IPMS MLLP Server Mock (Partial)

### PR DESCRIPTION
**Pull request by marrouchi**
## Type of PR (check all applicable)

- [x] 💼 New Package
- [ ] ♻️ Refactor
- [ ] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🧪 Tests
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

IPMS MLLP Server Mock : Still needs to be adjusted in the future in order to mimic the IPMS behavior.

## Added to documentation?

- [x] 📜 readme
- [ ] 📓 general documentation
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by BMartinos — APPROVED** [review]
Looks good to me :+1:

File: projects/ipms-mock/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+COPY . .
+
+# Install any needed packages specified in requirements.txt
+RUN pip install --no-cache-dir hl7apy watchdog
+
+# Make port 2575 available to the world outside this container
+EXPOSE ${SERVER_PORT}
+
+# Run app.py when the container launches
+CMD ["python", "app.py"]
```
Comment (ItsMurumba) [inline projects/ipms-mock/Dockerfile:25]:
```suggestion
CMD ["python", "app.py"]

```

File: projects/ipms-mock/.env.sample [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Environment settings for HL7 Mock Server
+SERVER_DOMAIN=0.0.0.0
+SERVER_PORT=2575
```
Comment (ItsMurumba) [inline projects/ipms-mock/.env.sample:3]:
```suggestion
SERVER_PORT=2575

```

File: projects/ipms-mock/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+```bash
+docker build -t hl7-mock-server .
+```
+Pass arguments: 
+
+```bash
+docker build --build-arg SERVER_DOMAIN_ARG=localhost --build-arg SERVER_PORT_ARG=2575 -t hl7-mock-server .
+```
+
+
```
Comment (ItsMurumba) [inline projects/ipms-mock/README.md:26]:
```suggestion
```

File: projects/ipms-mock/app.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    try:
+        while True:
+            pass
+    except KeyboardInterrupt:
+        observer.stop()
+        if initial_process:
+            initial_process.terminate()
+            initial_process.wait()
+    observer.join()
```
Comment (ItsMurumba) [inline projects/ipms-mock/app.py:41]:
```suggestion
    observer.join()
    
```

File: projects/ipms-mock/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        SERVER_DOMAIN_ARG: ${SERVER_DOMAIN}
+        SERVER_PORT_ARG: ${SERVER_PORT}
+    environment:
+      SERVER_DOMAIN: ${SERVER_DOMAIN}
+      SERVER_PORT: ${SERVER_PORT}
+    volumes:
+      - ./:/usr/src/app
+    ports:
+      - "${SERVER_PORT}:${SERVER_PORT}"
+    restart: unless-stopped
```
Comment (ItsMurumba) [inline projects/ipms-mock/docker-compose.yml:18]:
```suggestion
    restart: unless-stopped
    
```

File: projects/ipms-mock/server.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+logging.basicConfig(level=logging.DEBUG,
+                    format='%(asctime)s - %(levelname)s - %(message)s')
+
+# Environment variables for server configuration
+SERVER_DOMAIN = os.getenv('SERVER_DOMAIN', 'localhost')
+SERVER_PORT = int(os.getenv('SERVER_PORT', '2575'))
+
+_ROOT_PATH = os.path.dirname(os.path.abspath(__file__))
+
+
```
Comment (ItsMurumba) [inline projects/ipms-mock/server.py:21]:
```suggestion
```

File: projects/ipms-mock/server.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        return response
+
+    def create_error_response(self, error_msg):
+        logging.warning("Creating error response: %s", error_msg)
+        response = self.create_ack_response()
+        response.msa.msa_1 = 'AE'  # Error acknowledgment code
+        response.err.err_1 = error_msg
+        return response.to_mllp()
+
+
```
Comment (ItsMurumba) [inline projects/ipms-mock/server.py:87]:
```suggestion
```

File: projects/ipms-mock/server.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        'ORU^R01': (HL7Handler,),
+        'ADT^A01': (HL7Handler,),  # Admit a patient
+        'ADT^A02': (HL7Handler,),  # Transfer a patient
+        'ADT^A03': (HL7Handler,)   # Discharge a patient
+    }
+
+    # Set up and start the MLLP server
+    logging.info("Starting MLLP server on %s:%s", SERVER_DOMAIN, SERVER_PORT)
+    server = MLLPServer(SERVER_DOMAIN, SERVER_PORT, handlers)
+    server.serve_forever()
```
Comment (ItsMurumba) [inline projects/ipms-mock/server.py:101]:
```suggestion
    server.serve_forever()
    
```

**Review by ItsMurumba — CHANGES_REQUESTED** [review]
LGTM. Just some formatting changes suggestions.

**Pull request merged**